File: server/src/main/java/com/fluffy/submission/domain/SubmissionRepositoryCustom.java
package com.fluffy.submission.domain;

import com.fluffy.submission.domain.dto.MySubmissionSummaryDto;
import com.fluffy.submission.domain.dto.SubmissionSummaryDto;
import java.util.List;

public interface SubmissionRepositoryCustom {

    List<SubmissionSummaryDto> findSubmissionSummariesByExamId(Long examId);

    List<MySubmissionSummaryDto> findMySubmissionSummaries(Long examId, Long memberId);
}

File: server/src/main/kotlin/com/fluffy/submission/domain/SubmissionRepositoryCustom.kt
package com.fluffy.submission.domain

import com.fluffy.submission.domain.dto.MySubmissionSummaryDto
import com.fluffy.submission.domain.dto.SubmissionSummaryDto

interface SubmissionRepositoryCustom {

    fun findSubmissionSummariesByExamId(examId: Long): List<SubmissionSummaryDto>

    fun findMySubmissionSummaries(examId: Long, memberId: Long): List<MySubmissionSummaryDto>
}